src/main/java/org/apache/datasketches/memory/internal/package-info.java

```diff
@@ -0,0 +1,173 @@
+ * <blockquote><pre>
+ *     //Using try-with-resources block:
+ *     try (WritableyMapHandle handle = WritableMemoryImpl.map(File file)) {
+ *       WritableMemoryImpl wMem = handle.get();
+ *       doWork(wMem) // read and write to memory mapped file.
+ *     }
+ *
+ *     //Using explicit close():
+ *     WritableMapHandleImpl handle = WritableMemoryImpl.map(File file);
+ *     WritableMemoryImpl wMem = handle.get();
+ *     doWork(wMem) // read and write to memory mapped file.
+ *     handle.close();
+ * </pre></blockquote>
+ *
+ * <p>Where it is desirable to pass ownership of the resource (and the {@code close()}
+ * responsibility) one can not use the TWR block. Instead:</p>
+ * <blockquote><pre>
+ *     WritableMapHandleImpl handler = WritableMemoryImpl.map(File file);
+ *     doWorkAndClose(handle); //passes the handle to object that closes the resource.
+ * </pre></blockquote>
+ *
+ * <p>Whatever part of your process is responsible for allocating a resource external
+ * to the JVM must be responsible for closing it or making sure it gets closed.
+ * Since only the implementation Handles implement AutoCloseable, you must not let go of the
+ * handle reference until you are done with its associated resource.</p>
+ *
+ * <p>As mentioned above, there are two ways to do this:</p>
+ * <ul><li>Use a try-with-resources block.  At the end of the block, the JVM will automatically
+ * close the resource.</li>
+ *
+ * <li>If you need to pass an external resource, pass the implementation resource handle, not the
+ * access API. This means you are also passing the responsibility to close the resource.
+ * If you have different parts of your code holding references to the same handle,
+ * whichever one closes it first will make all the other resources invalid, so be careful.
+ * As long as there is at least one reference to the handle that is still valid and the resource
+ * has not been closed, the resource will remain valid. If you drop all references to all handles,
+ * the JVM will eventually close the resource, making it invalid, but it is possible that you might
+ * run out of memory first. Depending on this is a bad idea and a could be a serious,
+ * hard-to-find bug.</li>
+ * </ul>
+ *
+ *<p>Moving back and forth between <i>MemoryImpl</i> and <i>BufferImpl</i>:</p>
+ *<blockquote><pre>
+ *    MemoryImpl mem = ...
+ *    BufferImpl buf = mem.asBuffer();
+ *    ...
+ *    MemoryImpl mem2 = buf.asMemory();
+ *    ...
+ * </pre></blockquote>
+ *
+ * <p>Hierarchical memory regions can be easily created:</p>
+ * <blockquote><pre>
+ *     WritableMemoryImpl wMem = ...
+ *     WritableMemoryImpl wReg = wMem.writableRegion(offset, length); //OR
+ *     MemoryImpl reg = wMem.region(offset, length);
+ * </pre></blockquote>
+ *
+ * <p>With asserts enabled in the JVM, all methods are checked for bounds and
+ * use-after-close violations.</p>
+ *
+ * @author Lee Rhodes
+ */
+package org.apache.datasketches.memory.internal;
```

Comment:
There is another package info file in the root package that does not contain this documentation - did you intend to put this version in the internal package-info.java?

Reply:
Move the docs, thanks!